Use Hash in evalInVm to avoid flooding ram

FaeyUmbrea opened the pull request:
evalInVm caches every call with its script.
This replaces that with the hash value of the script instead, preventing resolveLocalCalls from flooding the ram with the script, causing out of memory errors on larger files.

BenBaryoPX commented:
This is a great suggestion :)
I think that we can use the already implemented [`generateScriptHash`](https://github.com/PerimeterX/restringer/blob/main/src/modules/utils/generateScriptHash.js) to the same effect, without adding a new dependency.
What do you think?

FaeyUmbrea commented:
Oh absolutely! I completely missed that function. 
Any hash with sufficiently large output and sufficiently low collision rate should work. 
I have implemented the change and added it to this PR.

BenBaryoPX commented on src/modules/unsafe/evalInVm.js:
```diff
@@ -3,6 +3,7 @@ const {badValue} = require(__dirname + '/../config');
 const defaultLogger = require(__dirname + '/../utils/logger');
 const getObjType = require(__dirname + '/../utils/getObjType');
 const createNewNode = require(__dirname + '/../utils/createNewNode');
+const hash = require(__dirname + '/../utils/generateScriptHash')
```
Looks great! Just to keep the style consistent, please import the name of the function as the name of the file, i.e. `generateScriptHash`.

FaeyUmbrea replied:
Done and done!

BenBaryoPX commented:
Thank you for taking the time to improve this tool :)
